Add option to allow non-OSBS build parent images

athos-ribeiro opened the pull request:
With the allow_external_base_image option set to True, do not stop
(fail) builds when a parent image is not found in Koji. Instead, raise a
working, skip digests checking and continue.

Signed-off-by: Athos Ribeiro <athos@redhat.com>

rcerven commented on atomic_reactor/plugins/pre_koji_parent.py:
```diff
@@ -103,8 +105,12 @@ def run(self):
-                raise RuntimeError(err_msg)
+                if get_allow_external_base_image(self.workflow, fallback=False):
+                    self.log.warning(err_msg)
+                    del self._parent_builds[img]
```
I don't think we want to remove it

in koji metadata : image->prebuild_build_id   and image->parent_image_builds
should still be something, None for build_id and just key with imagename used with {} as value for parent_image_builds
so it is clear that really doesn't have any koji build

rcerven commented on atomic_reactor/schemas/config.json:
```diff
@@ -441,6 +441,10 @@
         "description": "Proxy to access yum repositories",
         "type": "string"
     },
+    "allow_external_base_image": {
```
maybe "skip_koji_check_for_base_image" ?

external base image sound like we are using some image from some external repo

athos-ribeiro replied:
changed

twaugh commented:
'raise a working' → 'raise a warning'

MartinBasti commented:
Commit message misses issue ID

twaugh commented on atomic_reactor/schemas/config.json:
```diff
@@ -441,6 +441,10 @@
         "type": "string"
     },
+    "skip_koji_check_for_base_image": {
+        "description": "Allow base image without related koji build",
```
Could add an ` (insecure)` warning to the end of that maybe?

athos-ribeiro commented:
Refreshing review requests. I fixed all the previous requests and rebased the PR on master.

For documentation changes, [osbs-docs](https://osbs.readthedocs.io/en/latest/build_parameters.html#reactor-configuration) points to atomic-reactor config.json (so it is always up-to-date).

Any suggestions on where we should document the [koji metadata/registry] manifest digests verification?

MartinBasti commented:
Maybe we can add note here https://osbs.readthedocs.io/en/latest/users.html#koji-nvr about parent koji NVR requirement?

athos-ribeiro commented:
https://github.com/projectatomic/osbs-docs/pull/75

athos-ribeiro commented:
Release notes added:

- Add skip_koji_check_for_base_image config option to allow non-OSBS built images to be used as parent images
